exploration/text-vs-code.md

```diff
@@ -136,16 +209,93 @@ The current syntax includes some plain-ascii keywords:
+in those outer formats to aid human comprehension (e.g., depending upon container
+format, a U+000A LINE FEED might be represented as `\n`, `\012`, `\x0A`, `\u000A`,
+`\U0000000A`, `&#xA;`, `&NewLine;`, `%0A`, `<LF>`, or something else entirely).
+
+## Simple Messages
+
+In the Subcommittee meetings following Github discussions on Issues #493 and #499,
+the general consensus that formed for simple messages
+is that we allow them to be unquoted.
+
+("Simple messages" refers to messages consisting solely of a pattern, and thus are not complex messages.)
+
+Because the simple message pattern consists of the entire message,
+the pattern includes any leading or trailing whitespace.
+
+Given simple messages already being decided at a high level,
+the design decisions below for the proposed and alternative designs pertain specifically to complex messages.
 
 ## Proposed Design
 
+### Start in text, encapsulate message, always quote patterns
+
+Description:
+
+Since simple messages are unquoted (starting in text mode),
+complex messages must also start in text mode.
+
+Within a complex message, patterns are always quoted with `{{...}}` or other choice of delimiter.
+
+The entire complex message is also wrapped with `{{...}}` or other choice of delimiter.
+This allows interior "code mode" of message to have flexible whitespace in between tokens
+and _around_ quoted patterns.
+
+Pros
+
+* The rule about the whether leading and trailing whitespace is included is simple and unambiguous.
+* This matches the WYSIWIG behavior that simple messages preserve.
+* The patterns can be detected within the pattern more easily due to the delimiters serving as a visual anchor.
+* Requiring all patterns to be quoted minimizes the number of characters that need to be escaped within a pattern to 3:
+the 2 pattern delimiter characters and the escape character itself.
+* Because the sum of counts of declarations + `match` statement + `when` statements is always
+greater than or equal to the number of patterns,
+wrapping the entire message once yields less visual noise of repetitive code mode introducer symbols
+when there is 1+ declarations in a `match` (selection) message,
+or when there are 2+ declarations in a non-`match` complex message.
+
+Cons:
+
+* This comes at the cost of an inconsistency in the WYSIWYG patterns are quoted between simple and complex messages.
+In the case of simple messages, the containing format itself implicitly defines the beginning and end of the pattern (example: `"..."`), which is not visible at the level of MF2,
+while complex messages use the aforementioned delimiter to quote patterns (ex: `{{...}}`).
+* Another potential drawback, specifically in the case of non-`match` complex messages with exactly 1 declaration,
+is that this option adds 2 extra delimiters compared to an alternative syntax that doesn't require quoted patterns
+and is designed to minimize delimiter usage only to code mode introducers.
```

Comment:
We should add some of the other previously discussed drawbacks:

* If the code block is delimited from both sides, users may be tempted to insert text around it.
* If the code block is delimited from both sides, it may be easy to forget the `}}` closing the entire block.
* If we use curlies for patterns and for placeholders, then they serve double duty, which may make the syntax harder to understand, and also harder to make the pattern out visually.

Reply:
adding text after closing is not necessarily a drawback (I have a comment somewhere)

> If the code block is delimited from both sides, it may be easy to forget the }} closing the entire block.

Fair enough.
Maybe make it optional?

I think that `{{` is to make this intentionally ugly. I would probably go with `{#` to enter code mode, and `{` to wrap the patterns.
And yes, it is double douty. But choosing anything else  means that we escape one extra thing.
If we use `<<<` and I want (for some reason `<<< Hello world` as simple text, then we need to escape it.

So we add another escaping rule.
Pros and cons :-)

Reply:
>> If the code block is delimited from both sides, it may be easy to forget the }} closing the entire block.
> Forgetting to type syntax can happen in any alternative option, so this feels like a weaker argument than the previous.

I disagree a bit. Any enclosing syntax is prone to errors, but the argument here is that the more levels of nesting (and the further apart the enclosure bits), the more opportunities for error exist because the user is keeping track of more things.

Note that one of the proposals for "2a" was to have just a starter sigil. This would eliminate the enclosure for the message.